docs/source/developer/contributing.rst

```diff
@@ -352,19 +359,48 @@ With Homebrew:
 
    brew install pkg-config cairo pango libpng jpeg giflib librsvg
 
+.. _automatic_local_dev_env:
+
 Using automation to set up a local development environment
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-While there is a lot to learn by following the steps above, they can be automated to save time. The main advantages of using automation are: reduced time to get the environment up-and-running, reduced time to
+While there is a lot to learn by following the steps above, they can be automated to save time.
+The main advantages of using automation are: reduced time to get the environment up-and-running, reduced time to
 re-build the environment, better standardisation ("baseline", reproducible environments).
-This section shows how to do that using Docker and Vagrant.
+This section shows how to do that using VS Code dev containers, Docker and Vagrant.
+
+**Setup for VS Code**
```

Comment:
Maybe there could also be a section on using GitHub Codespaces, since they should also support devcontainers?

Reply:
done in [1495ad2](https://github.com/jupyterlab/jupyterlab/pull/15909/commits/1495ad235216bc17ec9d0899aaea07f9aaf1076b)